tuned/exports/unix_socket_exporter.py

```diff
@@ -0,0 +1,228 @@
+
+	def start(self):
+		if self.running():
+			return
+
+		self.stop()
+		self._construct_socket_object()
+
+	def stop(self):
+		if self._socket_object:
+			self._socket_object.close()
+
+	def _send_data(self, s, address, type, data):
+		"""
+		Sends data via socket s to address.
+		First byte of message specifies type of message (currently "r" for response, "e" for error message and "s" for signal).
+		Four more bytes are integer in network/big-endian with length of rest of message.
+		Rest of message is json data.
+		"""
+		log.debug("Sending socket data (type: %s, data: %s) to %s" % (type, data, address))
+		try:
+			j = json.dumps(data).encode("utf-8")
+			s.sendto(struct.pack("!sL", type.encode("utf-8"), len(j)), address)
+			s.sendto(j, address)
+		except Exception as e:
+			log.warning("Failed send data '%s' to '%s' type '%s': %s" % (data, address, type, e))
+
+	def period_check(self):
+		"""
+		Periodically checks socket object for new calls. This allows to function without special thread.
+		First four bytes is expected to be integer in network/big-endian with length of rest of message.
+		Rest of message expects json array, where first item is name of called method, rest is arguments of given method.
+		For response see method _send_data.
+		
+		Example implementation:
+		
+		def sss(d):
+			ssock_file = "/tmp/tuned_socket"
+			csock_file = "/tmp/tuned_socket_client"
+			if os.path.exists(csock_file):
+				os.remove(csock_file)
+			s=socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+			s.bind(csock_file)
+			j = json.dumps(d).encode("utf-8")
+			try:
+				s.sendto(struct.pack("!L", len(j)), ssock_file)
+				s.sendto(j, ssock_file)
+				d = s.recvfrom(5)
+				t, l = struct.unpack("!sL", d[0])
+				print("TYPE: %s" % t)
+				print("LEN: %s" % l)
+				data = json.loads(s.recvfrom(l)[0])
+				print("DATA: %s" % data)
+			except Exception as e:
+				print("E: %s" % e)
+			s.close()
+			
+		sss(("profiles",))
+		sss(("switch_profile", "balanced"))
+		"""
+		r, _, _ = select.select([self._socket_object], (), (), 0)
+		if r:
+			try:
+				l, address = self._socket_object.recvfrom(BYTES_LEN_OF_MSG)
+				l = struct.unpack("!L", l)[0]
+				log.debug("%s %s" % (address, l))
+			except Exception as e:
+				log.error("Failed to load length of message: %s" % e)
+				return
+			try:
+				data, address = self._socket_object.recvfrom(l)
+				log.debug("%s %s" % (address, data))
+			except Exception as e:
+				log.error("Failed to load data of message: %s" % e)
+				return
+			if not address:
+				log.warning("No return address for event, skipping")
+				return
+			if data:
+				try:
+					log.debug(data.decode("utf-8"))
+					data = json.loads(data.decode("utf-8"))
+				except Exception as e:
+					log.error("Failed to load json data '%s': %s" % (data, e))
+					self._send_data(self._socket_object, address, "err", "Failed to load json data '%s': %s" % (data, e))
+					return
+				if type(data) not in (tuple, list) or len(data) < 1:
```

Comment:
The protocol must be a bit more complex than this to be fully reliable. The client has to provide an identifier (random or whatever) with every call to be able to identify the correct answer since it is datagram based.

I suggest you take a look at existing standards like json-rpc here: https://en.wikipedia.org/wiki/JSON-RPC#Version_2.0

It is simple enough and has all the necessary structure to support future development.